package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +554,79 @@ def __init__(self, atomgroup, reference=None, select='all',
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
 
-        # Explicitly check for "mass" because this option CAN
-        # be used with groupselection. (get_weights() returns the mass array
-        # for "mass")
-        if not iterable(self.weights) and self.weights == "mass":
-            pass
-        else:
-            self.weights = get_weights(self.mobile_atoms, self.weights)
 
-        # cannot use arbitrary weight array (for superposition) with
-        # groupselections because arrays will not match
-        if (len(self.groupselections) > 0 and (
-                iterable(self.weights) or self.weights not in ("mass", None))):
-            raise ValueError("groupselections can only be combined with "
-                             "weights=None or weights='mass', not a weight "
-                             "array.")
+        # check weights type
+        #print(self.mobile_atoms)
+        if str(self.weights) == 'None':
```

Comment:
Why are you converting this into a string? You can also use `if self.weights is None:`?

Reply:
I thought it might have an impact when the comparison is elementwise in the future when I fix the FutureWarning for `self.weights == 'mass'`...ofc I was wrong.